services/spar/src/Spar/App.hs

```diff
@@ -453,23 +438,25 @@ instance
             runFinal $
               embedToFinal @IO $
                 randomToIO $
-                  runError @SparError $
-                    ttlErrorToSparError $
-                      ReaderEff.runReader (sparCtxOpts ctx) $
-                        galleyAccessToHttp (sparCtxLogger ctx) (sparCtxHttpManager ctx) (sparCtxHttpGalley ctx) $
-                          brigAccessToHttp (sparCtxLogger ctx) (sparCtxHttpManager ctx) (sparCtxHttpBrig ctx) $
-                            interpretClientToIO (sparCtxCas ctx) $
-                              samlUserStoreToCassandra @Cas.Client $
-                                idPToCassandra @Cas.Client $
-                                  defaultSsoCodeToCassandra $
-                                    scimTokenStoreToCassandra $
-                                      scimUserTimesStoreToCassandra $
-                                        scimExternalIdStoreToCassandra $
-                                          aReqIDStoreToCassandra $
-                                            assIDStoreToCassandra $
-                                              bindCookieStoreToCassandra $
-                                                runExceptT $
-                                                  runReaderT action ctx
+                  loggerToTinyLog (sparCtxLogger ctx) $
+                    mapLogger @String TinyLog.msg $
```

Comment:
I would move `mapLogger` into the interpreter (or wrapping a more convenient interpreter around the one you have, of course).  Then this here would be a little less mind-bending.